pkg/deploy/deploy.go

```diff
@@ -226,3 +231,26 @@ func waitForOpniStack(ctx context.Context, sc *Context) {
 		}
 	}
 }
+
+func disabled(asset string, disabledItems []string) bool {
+	for _, disabled := range disabledItems {
+		if strings.Contains(asset, disabled) {
+			return true
+		}
+	}
+	return false
+}
+
+func waitForNS(ctx context.Context, sc *Context, ns string) {
+	ticker := time.NewTicker(10 * time.Second)
```

Comment:
Suggest using the `wait` package in k8s.io/apimachinery for this. [godoc](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait)
Example:
```go
c, cancel := context.WithCancel(ctx)
wait.UntilWithContext(c, func() {
  if _, err := sc.K8s.CoreV1().Namespaces().Get(c, ns, metav1.GetOptions{}); err == nil {
    cancel()
  }
}, 1 * time.Second)
```
This will allow you to quit out of the polling loop when the context is canceled. 

You may also want to check if the error you get back from the k8s api is a "does not exist" error so you can properly report other error kinds i.e. connection failure

Reply:
will fix, thanks